pkg/snapshot/snapshotter/garbagecollector.go

```diff
@@ -216,6 +222,10 @@ func (ssr *Snapshotter) GarbageCollectChunks(snapList brtypes.SnapList) (int, br
 		}
 		// delete the chunk object
 		snapPath := path.Join(snap.SnapDir, snap.SnapName)
+		if !snap.IsDeletable() {
+			ssr.logger.Infof("GC: Skipping : %s, since it is immutable", snap.SnapName)
```

Comment:
I would prefer if we do not use the phrase "retention period" since we already use this phrase when talking about delta snapshot retention period, which is not related to immutability.

Using this phrase to talk about immutability would be overloading this phrase for two different features. I would suggest we use "immutability period", instead.

Do you have any reservations @ishan16696?

Reply:
ok, but isn't both are of same functionality or am I missing something ?

Reply:
Retention period of deltas concerns with how long delta snapshots are persisted before they are considered for garbage collection.
https://github.com/gardener/etcd-backup-restore/blob/7e78860841fca85b2ced4dea080f4df65806ebb4/docs/usage/garbage_collection.md?plain=1#L25-L27
Immutability period is the period for which every object that is uploaded to the object store is immutable.

For example, delta snapshot retention period could be set to 6 days, with immutability period to 2.

Immutability period has to be lesser than or equal to the delta snapshot retention period.

Or am I missing something @ishan16696?

Reply:
I think there is a ambiguity in terminology. I'm referring `retention period` as `retention period of object/bucket lock` where as I guess you are referring to `retention period` as retention of backups w.r.t garbage collection policy.

Reply:
Yes, there seems to be confusion here.

I propose that we continue to use the word retention for `retention period of delta snapshots` which corresponds with garbage collection, and use `immutability period` which corresponds with `object/bucket lock`.

Do you have any opinion on this naming scheme?

Reply:
> I propose that we continue to use the word retention for retention period of delta snapshots which corresponds with garbage collection, and use immutability period which corresponds with object/bucket lock.

yeah, make sense to me

Reply:
Addressed in [a6680d4](https://github.com/gardener/etcd-backup-restore/pull/776/commits/a6680d4ad3e2247f52374b5423b2ada4016e6b1c).